pinot-common/src/main/java/org/apache/pinot/sql/parsers/CalciteSqlParser.java

```diff
@@ -95,7 +95,7 @@ private CalciteSqlParser() {
   //   `OPTION (<k1> = <v1>) OPTION (<k2> = <v2>) OPTION (<k3> = <v3>)`
   private static final Pattern OPTIONS_REGEX_PATTEN =
       Pattern.compile("option\\s*\\(([^\\)]+)\\)", Pattern.CASE_INSENSITIVE);
-
+  private static final Pattern COMMENTED_QUERY_PATTERN = Pattern.compile("-{2,}");
```

Comment:
Can use simply search for "--" instead of using regex match? Substring search is much cheaper comparing to regex match

Reply:
This is done – our original problem with using `--` pattern as signpost remains.